File: back/prod/src/models/equipments.py
from __future__ import annotations

from typing import List  # noqa: UP035

from pydantic import BaseModel, ConfigDict


class Equipment(BaseModel):
    """Model for a kitchen equipment."""

    model_config = ConfigDict(extra="forbid")
    id: str
    name: str
    quantity: float

class Equipments(BaseModel):
    """Model for a list of kitchen equipments."""

    equipments: List[Equipment]  # noqa: UP006

File: front/nextjs_ts/src/types/equipments.ts
export interface Equipment {
    id: string;
    name: string;
    quantity: number;
};

export interface Equipments {
    equipments: Equipment[];
};

export interface EquipmentsContextType {
    equipments: Equipments | null;
    setEquipments: (equipments: Equipments | null) => void;
    isLoading: boolean;
    error: string | null;
};